File: tyama_codeeval52.rb
#!/usr/bin/ruby
A=['']+%w[One Two Three Four Five Six Seven Eight Nine Ten Eleven Twelve Thirteen Fourteen Fifteen Sixteen Seventeen Eighteen Nineteen]
B=%w[Twenty Thirty Forty Fifty Sixty Seventy Eighty Ninety]
def txt(x)
	s=''
	s+=A[x/100]+'Hundred' if x/100>0
	return s+A[x%100] if x%100<20
	s+B[x/10%10-2]+A[x%10]
end
while gets
	x=$_.to_i
	y=x/1000000000
	print txt(y)+'Billion' if y>0
	y=x/1000000%1000
	print txt(y)+'Million' if y>0
	y=x/1000%1000
	print txt(y)+'Thousand' if y>0
	y=x%1000
	print txt(y) if y>0
	puts 'Dollars'
end

File: tyama_codeeval52.cpp
#include <string>
#include <cstdio>
using namespace std;
string A[]={
	"",
	"One","Two","Three","Four","Five","Six","Seven","Eight","Nine","Ten",
	"Eleven","Twelve","Thirteen","Fourteen","Fifteen","Sixteen","Seventeen","Eighteen","Nineteen"
};
string B[]={"Twenty","Thirty","Forty","Fifty","Sixty","Seventy","Eighty","Ninety"};
string txt(int x){
	string s;
	if(x/100)s+=A[x/100]+"Hundred";
	if(x%100<20)return s+A[x%100];
	return s+B[x/10%10-2]+A[x%10];
}
main(){
	int x,y;
	for(;~scanf("%d",&x);){
		//y=x/1000000000;
		//if(y)printf((txt(y)+"Billion").c_str());
		y=x/1000000%1000;
		if(y)printf((txt(y)+"Million").c_str());
		y=x/1000%1000;
		if(y)printf((txt(y)+"Thousand").c_str());
		y=x%1000;
		if(y)printf(txt(y).c_str());
		puts("Dollars");
	}
}